src/libcore/fmt/mod.rs

```diff
@@ -1718,6 +1725,7 @@ macro_rules! tuple {
         #[stable(feature = "rust1", since = "1.0.0")]
         impl<$($name:Debug),*> Debug for ($($name,)*) where last_type!($($name,)+): ?Sized {
             #[allow(non_snake_case, unused_assignments, deprecated)]
+            #[inline]
             fn fmt(&self, f: &mut Formatter) -> Result {
```

Comment:
This is an example of one we don't need to mark inline. We want to mark those that are not generic and forwarding directly to another non-generic function as inline.

Reply:
Thanks! With those conditions, should the only ones that I mark inline be non-float numbers, the unit type and bools? Just making sure.

Reply:
Yes, I agree.

I'm not 100% sure which trade off is the best for the function pointers, ideas welcome.

Reply:
I'd even skip the unit type if pad is a method that inlines. We don't want to inline a bigger function with this PR.

Reply:
The unit type's pad method doesn't inline, so I think I'll inline the unit type's fmt function. For function pointers, I think I'll avoid marking them as inline them for now.